colpali_engine/collators/corpus_query_collator.py

```diff
@@ -0,0 +1,78 @@
+from random import randint
+from typing import Any, Dict, List, Optional, cast
+
+from datasets import Dataset
+
+from colpali_engine.collators.visual_retriever_collator import VisualRetrieverCollator
+from colpali_engine.utils.processing_utils import BaseVisualRetrieverProcessor
+
+
+class CorpusQueryCollator(VisualRetrieverCollator):
+    def __init__(
+        self,
+        processor: BaseVisualRetrieverProcessor,
+        max_length: int = 2048,
+        image_dataset: Optional[Dataset] = None,
+        mined_negatives: bool = True,
+        corpus_format: str = "wikiss",
+    ):
+        super().__init__(
+            processor=processor,
+            max_length=max_length,
+        )
+        if image_dataset is None:
+            raise ValueError("`image_dataset` must be provided")
+        self.image_dataset = cast(Dataset, image_dataset)
+        self.mined_negatives = mined_negatives
+        self.corpus_format = corpus_format
+
+        if self.corpus_format == "wikiss":
+            print("Mapping docids to indices")
+            self.docid_to_idx = {docid: idx for docid, idx in
+                                 zip(self.image_dataset["docid"], range(len(self.image_dataset)))}
+
+    def get_image_from_docid(self, docid):
+        if self.corpus_format == "wikiss":
+            return self.image_dataset[self.docid_to_idx[docid]]["image"]
+        elif self.corpus_format == "docmatix":
+            doc_page = int(docid.split("_")[1])
+            doc_id = int(docid.split("_")[0])
+            return self.image_dataset[doc_id]["images"][doc_page]
+        elif self.corpus_format == "vidore":
+            return self.image_dataset[int(docid)]["image"]
+        else:
+            raise NotImplementedError(f"Corpus format {self.corpus_format} not supported")
```

Comment:
Quite elegant, I like it a lot! If possible, could you add the missing typing for `docid` (which I would renamed `doc_id`) please?